test/libsolidity/SemanticTest.cpp

```diff
@@ -288,8 +299,26 @@ TestCase::TestResult SemanticTest::runTest(
 		return TestResult::Failure;
 	}
 
-	if (!success && (m_runWithYul || !_compileViaYul))
+	if (success && !m_testCaseWantsEwasmRun && _isEwasmRun)
 	{
+		if (m_revertStrings != RevertStrings::Default)
+			return TestResult::Success;
+
+		string message = success ?
+						 "Test can pass via Yul (ewasm), but marked with \"compileToEwasm: false.\"" :
+						 "Test compiles via Yul (ewasm), but it gives different test results.";
```

Comment:
This seems to have been adjusted after https://github.com/ethereum/solidity/pull/10356#discussion_r565419306 but not entirely correctly. `success` will always be true in `string message = success ?` because the `if` at the top requires `success` to be true. You need to do it like in the block for Yul above.

Reply:
After looking into this again, I think it makes no sense to support the different-test-results-case. The problem is that we have a lot of these. In many cases Ewasm just returns a failure. So I think it makes sense just to inform on success, that the test-case could be updated to set `compileToEwasm` and just ignore different output for now.